UnicornEngine/video/source/vulkan/Renderer.cpp

```diff
@@ -877,21 +970,27 @@ bool Renderer::CreateRenderPass()
 bool Renderer::CreateDescriptionSetLayout()
 {
     std::vector<vk::DescriptorPoolSize> descriptorPoolSizes;
-    vk::DescriptorPoolSize descriptorMVPPoolSize;
-    descriptorMVPPoolSize.type = vk::DescriptorType::eUniformBuffer;
-    descriptorMVPPoolSize.descriptorCount = 1;
+
+    vk::DescriptorPoolSize descriptorViewProjectionPoolSize;
+    descriptorViewProjectionPoolSize.type = vk::DescriptorType::eUniformBuffer;
+    descriptorViewProjectionPoolSize.descriptorCount = 1;
 
     vk::DescriptorPoolSize descriptorModelPoolSize;
     descriptorModelPoolSize.type = vk::DescriptorType::eUniformBufferDynamic;
     descriptorModelPoolSize.descriptorCount = 1;
 
-    descriptorPoolSizes.push_back(descriptorMVPPoolSize);
+    vk::DescriptorPoolSize descriptorSamplerPoolSize;
+    descriptorSamplerPoolSize.type = vk::DescriptorType::eCombinedImageSampler;
+    descriptorSamplerPoolSize.descriptorCount = m_physicalDeviceProperties.limits.maxDescriptorSetSampledImages;
+
+    descriptorPoolSizes.push_back(descriptorViewProjectionPoolSize);
     descriptorPoolSizes.push_back(descriptorModelPoolSize);
+    descriptorPoolSizes.push_back(descriptorSamplerPoolSize);
 
     vk::DescriptorPoolCreateInfo poolCreateInfo;
     poolCreateInfo.poolSizeCount = static_cast<uint32_t>(descriptorPoolSizes.size());
     poolCreateInfo.pPoolSizes = descriptorPoolSizes.data();
-    poolCreateInfo.maxSets = 2;
+    poolCreateInfo.maxSets = 3000; //TODO: expand 
```

Comment:
please do

Reply:
Can't. Need to think about adaptive expansion. 3000 means here that we can support only 3000 
different textures in one scene.
Add annotation?